### PR TITLE
Reviewer Role

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR adds reviewer functionality to OpenCfP.

The reviewer currently has the following options
* See talks and speakers
* Rate talks

The reviewer can not comment on talks, favorite or select them, remove speakers or talks, promote or demote to/from admin. They also can't see the speakers airport.

In the future it would be good to let the admin decide what the reviewer can and can not do/ see. Either through config files, or another way.

Related to #415, #485 .

Changes in this PR
* Addition of reviewer role
* Ability to promote to/demote from reviewer through command line
* Basic refactor of Command classes
* promote function on AccountManagement is now promoteTo, and has a second optional parameter which is the role to promote to, defaults to admin
* The same for the demote function, which is now DemoteFrom
* Addition of a 'search' scope in the user model, which does a 'like' search
* getFilteredTalks has been moved to its own class
* 3 new controllers, for reviewers (Dashboard, Speaker, Talks)
* AdminAccess is now RoleAccess, and takes the role to check for as an argument
* Updated to the UserAccess interface to make that possible
* New Routes/ Providers/ views/ tests
* Reviewer role Migration

I will be making a separate PR to move Pagination to its own space, which should fix some of the Code Climate issues, but thats outside of the scope for this PR, since its growing pretty big already.

### CODE REVIEW

File: tests/Http/Controller/Reviewer/DashboardControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+namespace OpenCFP\Test\Http\Controller\Reviewer;
+
+use OpenCFP\Test\WebTestCase;
+
+class DashboardControllerTest extends WebTestCase
+{
+    /**
+     * @test
+     */
+    public function indexActionWorksWithoutTalks()
```
Comment (BackEndTea) [inline tests/Http/Controller/Reviewer/DashboardControllerTest.php]: I can't quite figure out why this test is failing, as it works fine for me locally.

**Comment by chartjes** [comment]
@BackEndTea is this no longer WIP?

**Comment by BackEndTea** [comment]
@chartjes This is no longer WIP.
It is a good base for the reviewer role, though there are still features that could be created for it.

**Comment by chartjes** [comment]
@BackEndTea Once the test failures are fixed, I'm happy to merge this in and cut a v1.3 release with this functionality in it https://travis-ci.org/opencfp/opencfp/builds/300230058?utm_source=github_status&utm_medium=notification

**Comment by chartjes** [comment]
Thanks for this great new feature @BackEndTea

**Comment by BackEndTea** [comment]
Thanks @chartjes

**Pull request merged**